src/plugins/discover/public/application/main/components/sidebar/discover_sidebar_responsive.tsx

```diff
@@ -180,30 +199,18 @@ export function DiscoverSidebarResponsive(props: DiscoverSidebarResponsiveProps)
   const canEditDataView =
     Boolean(dataViewEditor?.userPermissions.editDataView()) || !selectedDataView?.isPersisted();
 
-  useEffect(
-    () => {
-      // For an external embeddable like the Field stats
-      // it is useful to know what fields are populated in the docs fetched
-      // or what fields are selected by the user
-
-      const fieldCnts = fieldCounts.current ?? {};
+  useEffect(() => {
+    // For an external embeddable like the Field stats
+    // it is useful to know what fields are populated in the docs fetched
+    // or what fields are selected by the user
 
-      const availableFields = props.columns.length > 0 ? props.columns : Object.keys(fieldCnts);
-      availableFields$.next({
-        fetchStatus: FetchStatus.COMPLETE,
-        fields: availableFields,
-      });
-    },
-    // Using columns.length here instead of columns to avoid array reference changing
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [
-      selectedDataView,
-      availableFields$,
-      fieldCounts.current,
-      documentState.result,
-      props.columns.length,
-    ]
-  );
+    const availableFields =
+      props.columns.length > 0 ? props.columns : Object.keys(fieldCounts || {});
+    availableFields$.next({
+      fetchStatus: FetchStatus.COMPLETE,
+      fields: availableFields,
+    });
+  }, [selectedDataView, fieldCounts, props.columns, availableFields$]);
```

Comment:
Removing `availableFields$` could be a good issue to create for next steps, and to allow granular planning, since we should use `useExistingFieldsFetcher` also in our field stats